### ISSUE TITLE
Pair Losing Bought Price

### ISSUE THREAD

**Issue by hkhan24** [taniman/profit-trailer#2073]
NO NEW FEATURES! Use [PT Enahancements](https://github.com/taniman/profit-trailer-enhancements) for Feature Requests.

## ProfitTrailer info
Version:

 
Exchange:
- [X ] Bittrex
- [ ] Binance
- [ ] Poloniex

 
Market:
- [X ] BTC
- [ ] ETH
- [ ] USDT
- [ ] BNB
- [ ] XMR

 
Operating System:
- [ ] Windows (please specify version)
- [X ] Ubuntu (please specify version)
- [ ] Other linux distro (please specify)
- [ ] Mac OS

## Expected Behavior
All Pairs Should have a bought price

## Current Behavior
20% of pairs fail to have a bought price

## Troubleshooting
Restarted bot and manually set bought prices

## Steps to Reproduce

1.
2.
3.
4.

## Configuration
Using Spanish cross

## Log files/console output

 
![image](https://user-images.githubusercontent.com/6436539/45948157-d3b86980-bfee-11e8-80b8-d548e7fd145a.png)

![image](https://user-images.githubusercontent.com/6436539/45948192-e5017600-bfee-11e8-8a9f-a684f3f5e9b7.png)

**Comment by hkhan24** [comment]
Version 2.1.12

**Comment by taniman** [comment]
Please download and upload your log file here. 

For now you would need to set the prices manually. This looks like a problem caused by the latest bittrex changes that happened

**Comment by hkhan24** [comment]
[ptlog-24092018-153854.txt](https://github.com/taniman/profit-trailer/files/2411220/ptlog-24092018-153854.txt)

**Comment by hkhan24** [comment]
Befoe it was doing this:
https://user-images.githubusercontent.com/6436539/45926883-a2259c80-bf21-11e8-8b91-c835eee39400.png

But not seeing that anymore.

**Comment by cas8180** [comment]
I have been having this issue since 2.x I have seen multiple changelogs attempting to resolve this issue but it never ends up being resolved. I need to constantly monitor my bot(s) because of this issue and manually set prices when they get lost. I usually end up losing an opporunity to trade because of this and end up with a new bag. I believe with this bug still being there it defeats the point of trade automation. I think there should be multiple fallback stategies of a pair every loses its bought price. There should be a way to query the API and retrieve the original bought price OR write the original bought price to disk at the time of purchase, OR use the new cloud implementation that you guys have to store that original bought price to your remote servers. There should simply never ever be a case where a bought price is lost and we have to set it manually.

**Comment by taniman** [comment]
@cas8180 Dont hijack posts. 

I have over 500 sales last couple of weeks over multiple bots. not 1 price lost. 

Sure We could just put whatever price down as a failsafe. But that would be worse than making you set your price when you are doing something out of the ordinary or something out of the ordinary happens.

**Comment by cas8180** [comment]
What you purchased a coin for never changes. If you want to override because you want to do something custom that's fine. However from an automation perspective what you bought something for is always what you bought it for.


Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by taniman** [comment]
@cas8180 Now I am curious into all these prices u are losing as u seem to know about automation.
Please provide your logfile with 1 or 2 lost prices and then also provide the unaccounted folder files belonging to these and I will investigate

**Comment by hkhan24** [comment]
[ProfitTrailer.log](https://github.com/taniman/profit-trailer/files/2414490/ProfitTrailer.log)
Here we go. Hope that helps.

**Comment by hkhan24** [comment]
Upgrading to 2.1.13 now and I'll see how that goes at the suggestion of PTElroy

**Comment by taniman** [comment]
Yes, this is as I suspected. Problem with prices was caused when the pending orders you had were cancelled and an error occurred. Causing the merge to not be successful.

Please update to 2.1.13 to prevent this from happening in the future.

There is one other thing happening on bittrex currently. And issue was just created and  will be fixed in the next release hopefully. It does not happen all the time so I hope you do not encounter it. :)

**Comment by hkhan24** [comment]
Thank you so much for your help and quick resolution! I am now on 2.1.13 and hopefully all will be well :)

**Comment by cas8180** [comment]
Thank's Taniman. Attached are my logs as requested. In each case that you see a hot config set. That is where I had to set the purchase price manually because it was somehow lost. 

You could say I know a little about automation. Wrote a primitive bot just to learn the exchanges API's its listed under my profile on github. Also been doing software development professionally for 20 years. But hey that's neither here nor there.... :)

[ProfitTrailer.log](https://github.com/taniman/profit-trailer/files/2416842/ProfitTrailer.log)
